adafruit_mcp3xxx/differential_analog_in.py

```diff
@@ -0,0 +1,60 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2018 Brent Rubell for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`differential_analogin.py`
+======================================================
+Differential ADC read for mcp3xxx ADCs.
+
+* Author(s): Brent Rubell
+"""
+
+class DifferentialAnalogIn():
+    """Differential ADC read."""
+
+    def __getitem__(self, key):
+        return self._channels[self._pins[key]]
+
+    def __init__(self, mcp, pin_1, pin_2):
```

Comment:
I'd combine this with AnalogIn and add a `negative_pin` kwarg-only argument that is `None` by default for single ended mode. So, it'd be:
```python
def __init__(self, mcp, positive_pin, *, negative_pin=None):
```